src/shared/components/events-list/index.js

```diff
@@ -2,134 +2,119 @@
 // You can request only displaying events of past or future
 // with the `timeline` prop
 
-/* eslint-disable max-len */
-
-import React, { Component } from 'react';
+import React, { PropTypes } from 'react';
 import styles from './style.css';
 
 import EventImage from '../event-image';
-import { imageAssetsEndpoint } from '../../config';
 import DateBubble from '../date-bubble';
 import HR from '../hr';
 import { Grid, Cell } from '../grid';
-import classNames from 'classnames';
-import icons from '../icons/style.css';
 
 import TagsList from '../tags-list';
 import EventLinksList from '../event-links-list';
+import EventsTimelineTitle from '../events-timeline-title';
+import EventTitle from '../event-title';
 import { eventHref } from '../../util/event';
-import { splitEvents } from '../../util/split-events';
+import { splitEvents, setEndDate, eventImagePath } from '../../util/events';
 
-export default class EventsList extends Component {
-  static propTypes = {
-    events: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
-    timeline: React.PropTypes.oneOf(['past', 'future', 'today']),
-  };
+const EventsList = ({
+  events,
+  timeline,
+}) => {
+  const relevantEvents =
+    timeline
+    ? splitEvents({
+      events,
+      timeline,
+      reverse: timeline === 'future',
+    })
+    : events;
 
-  render() {
-    const relevantEvents =
-      this.props.timeline
-      ? splitEvents(
-        this.props.events,
-        this.props.timeline,
-        { reverse: this.props.timeline === 'future' }
-      )
-      : this.props.events;
+  if (relevantEvents.length > 0) {
+    return (
+      <div className={styles.eventsListTimelineSection}>
+          <EventsTimelineTitle timeline={timeline} />
+        <ul className={styles.eventsList}>
+          {
+            relevantEvents.map((event) => (
```

Comment:
This is quite a complex component. Could simplify it further here to have a child '<RelevantEvent ... />' etc. I think more complex tests are needed to ensure that all the ? checks achieve the desired result - seems easy to break